build.sbt

```diff
@@ -51,6 +51,14 @@ val jlineDeps         = Seq(jlineDep, jnaDep)
 val testInterfaceDep  = "org.scala-sbt"                  % "test-interface"                   % "1.0"
 val diffUtilsDep      = "com.googlecode.java-diff-utils" % "diffutils"                        % "1.3.0"
 
+// `set Global / fatalWarnings := true` to enable -Werror for the certain modules
+// currently, many modules cannot support -Werror; ideally this setting will eventually
+//   enable -Werror for all modules
+val fatalWarnings = settingKey[Boolean]("whether or not warnings should be fatal in the build")
```

Comment:
ideally such a setting could have a different name from the old compiler option. Maybe a metaphor from politics. `leaning := "conservative"` for strict or `"liberal"` to allow behaviors that raise red flags. Unless when compiling under `--covid`, in which case the semantics are exactly reversed.

Reply:
I could call it `werror`

Reply:
`werror` or not you do is fine with me.